Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -766,7 +766,17 @@ PTF_TEST_CASE(TestMtuSize)
 	// Construct a packet within the MTU and assert that it should send
 	// Source and destination addresses are somewhat arbitrary. Only important thing is that the packet is valid
 	pcpp::EthLayer smallEthernetLayer(liveDev->getMacAddress(), pcpp::MacAddress("aa:bb:cc:dd:ee:ff"));
-	pcpp::IPv4Layer smallIPLayer(ipToSearch, pcpp::IPv4Address(PcapTestGlobalArgs.remoteIp.c_str()));
+
+	pcpp::IPv4Address remoteIpAddress;
+	try
+	{
+		remoteIpAddress = pcpp::IPv4Address(PcapTestGlobalArgs.remoteIp.c_str());
+	}
+	catch (...)
+	{
+		remoteIpAddress = ipToSearch;
```

Comment:
Maybe the test should fail if there's an exception here?

Reply:
true, I forgot to add a try for the `ipToSearch` when it was constructed.

Reply:
fixed all `pcpp::IPv4Address ipToSearch`.

Reply:
I don't think we should catch an exception and fail the test. An exception will anyway fail the test. What I meant in this specific line is that we should catch an exception and just keep the code the way it was originally:

```cpp
pcpp::IPv4Layer smallIPLayer(ipToSearch, pcpp::IPv4Address(PcapTestGlobalArgs.remoteIp.c_str()));
```

Reply:
@seladb, If left unhandled, the test code will crash immediately. I suggest capturing it instead, as it will facilitate pinpointing the failure point when the test fails.

Reply:
@seladb I would like to know your opinion about this.

Reply:
sure, reverted.

and back to your original comment, we have the code:
```
	pcpp::PcapLiveDevice* liveDev = nullptr;
	pcpp::IPv4Address ipToSearch(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
	liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(ipToSearch);
```

in the beginning, so this line is safe.

Reply:
ah, I think I got your point, the reason we need to have try catch here is that `PcapTestGlobalArgs.remoteIp` may be empty, in the previous code, if the IP is invalid, it is set to zero by default, but now if the IP is invalid, it will crash directly. To make the behavior the same, we need to set the `remoteIpAddress` to a valid one in this case.

Reply:
It can be any valid IP, so I think `pcpp::IPv4Address::Zero` should be fine.